File: docs/guides/examples/py/src/buckets_create.py
from reduct import Client, BucketSettings, QuotaType

# Create a client with the base URL and API token
client = Client("http://localhost:8383", api_token="my-token")


async def create_bucket():
    # Create a bucket with the name "my-bucket" and a FIFO quota of 1GB
    settings = BucketSettings(
        quota_type=QuotaType.FIFO,
        quota_size=1000_000_000,
    )
    bucket = await client.create_bucket("my-bucket", settings, exist_ok=True)
    assert bucket.name == "my-bucket"


if __name__ == "__main__":
    import asyncio

    asyncio.run(create_bucket())

File: versioned_docs/version-1.12.x/examples/rs/examples/buckets_create.rs
use reduct_rs::{QuotaType, ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a new client with the API URL and API token
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();

    // Create a bucket with the name "my-bucket" and a FIFO quota of 1GB
    let bucket = client
        .create_bucket("my-bucket")
        .quota_type(QuotaType::FIFO)
        .quota_size(1_000_000_000)
        .exist_ok(true)
        .send()
        .await?;

    assert_eq!(bucket.name(), "my-bucket");
    Ok(())
}
